### PR TITLE
Fby3.5 hd support bom3 sensor

### PR DESCRIPTION
**Pull request by Yi-Shum**
Summary:
- Support VR sensor MP2856GUT
- Add the sensor config table of 3rd source sensor, replace sensor talbe to it if board revision is SYS_BOARD_EVT_BOM3.

Dependency: #467 

Test Plan:
- Build Code: Pass

Known issue:
- Pout reading for MP2856GUT is larger than expected, still checking with FAE.

Log:
Third source boar
```
root@bmc-oob:~# sensor-util slot1
...
HSC Temp                     (0xE) :   47.12 C     | (ok)
CPU0 VR Temp                 (0xF) :   49.00 C     | (ok)
SOC VR Temp                  (0x10) :   51.00 C     | (ok)
CPU1 VR Temp                 (0x11) :   34.00 C     | (ok)
PVDDIO VR Temp               (0x12) :   34.00 C     | (ok)
PVDD11 VR Temp               (0x13) :   31.00 C     | (ok)
P12V_STBY Vol                (0x14) :   12.00 Volts | (ok)
PVDD18_S5 Vol                (0x15) :    1.81 Volts | (ok)
P3V3_STBY Vol                (0x16) :    3.30 Volts | (ok)
PVDD11_S3 Vol                (0x17) :    1.10 Volts | (ok)
P3V_BAT Vol                  (0x18) :    3.08 Volts | (ok)
PVDD33_S5 Vol                (0x19) :    3.31 Volts | (ok)
P5V_STBY Vol                 (0x1A) :    4.99 Volts | (ok)
P12V_MEM_1 Vol               (0x1B) :   12.05 Volts | (ok)
P12V_MEM_0 Vol               (0x1C) :   12.03 Volts | (ok)
P1V2_STBY Vol                (0x1D) :    1.20 Volts | (ok)
P3V3_M2 Vol                  (0x1E) :    3.31 Volts | (ok)
P1V8_STBY Vol                (0x1F) :    1.81 Volts | (ok)
HSC Input Vol                (0x20) :   12.05 Volts | (ok)
CPU0 VR Vol                  (0x21) :    0.92 Volts | (ok)
SOC VR Vol                   (0x22) :    1.19 Volts | (ok)
CPU1 VR Vol                  (0x23) :    0.93 Volts | (ok)
PVDDIO VR Vol                (0x24) :    1.10 Volts | (ok)
PVDD11 VR Vol                (0x25) :    1.10 Volts | (ok)
HSC Output Cur               (0x26) :   12.75 Amps  | (ok)
CPU0 VR Cur                  (0x27) :   12.75 Amps  | (ok)
SOC VR Cur                   (0x28) :   59.75 Amps  | (ok)
CPU1 VR Cur                  (0x29) :   16.75 Amps  | (ok)
PVDDIO VR Cur                (0x2A) :   31.75 Amps  | (ok)
PVDD11 VR Cur                (0x2B) :    3.25 Amps  | (ok)
HSC Input Pwr                (0x2C) :  155.15 Watts | (ok)
CPU0 VR Pwr                  (0x2D) :   96.00 Watts | (ok)
SOC VR Pwr                   (0x2E) :  568.00 Watts | (ucr)
CPU1 VR Pwr                  (0x2F) :  120.00 Watts | (ok)
PVDDIO VR Pwr                (0x30) :  280.00 Watts | (ucr)
PVDD11 VR Pwr                (0x31) :   32.00 Watts | (ok)
...
```

### CODE REVIEW

File: common/shell/shell_platform.h [hunk 1]
```diff
@@ -67,16 +67,19 @@ const char *const sensor_type_name[] = {
...
-				ltc4282) sensor_name_to_num(fan) sensor_name_to_num(tmp431)
-				sensor_name_to_num(pmic) sensor_name_to_num(
-					ina233) sensor_name_to_num(isl69254)
-					sensor_name_to_num(max16550a) sensor_name_to_num(
-						ina230) sensor_name_to_num(xdpe12284c)
-						sensor_name_to_num(raa229621) sensor_name_to_num(
-							nct7718w) sensor_name_to_num(ltc4286)
-							sensor_name_to_num(amd_tsi)
-								sensor_name_to_num(apml_mailbox)
+			sensor_name_to_num(xdpe15284) sensor_name_to_num(ltc4282) sensor_name_to_num(
```
Comment (garnermic) [inline common/shell/shell_platform.h:70]: Please reformat to make this section easier to read.

File: meta-facebook/yv35-hd/src/platform/plat_class.c [hunk 2]
```diff
@@ -296,5 +306,44 @@ void init_platform_config()
...
+	/* Set HSC OCP */
+	uint8_t ilim_adjust = ILIM_ADJUST_DEFAULT;
+	if (_2ou_status.present) {
+		if (hsc_module == HSC_MODULE_ADM1278) {
+			gpio_set(HSC_OCP_GPIO1_R, GPIO_LOW);
+			gpio_set(HSC_OCP_GPIO2_R, GPIO_LOW);
+			gpio_set(HSC_OCP_GPIO3_R, GPIO_HIGH);
+		} else if (hsc_module == HSC_MODULE_LTC4282) {
+			ilim_adjust = ILIM_ADJUST_2OU_EXP;
+		}
```
Comment (garnermic) [inline meta-facebook/yv35-hd/src/platform/plat_class.c:318]: Please add an else to capture invalid hsc_module value.

File: meta-facebook/yv35-hd/src/platform/plat_class.c [hunk 3]
```diff
@@ -296,5 +306,44 @@ void init_platform_config()
...
+	} else if (_1ou_status.present && (_1ou_status.card_type != TYPE_1OU_RAINBOW_FALLS)) {
+		printf("Unsupported configuration, 1ou card type %d\n", _1ou_status.card_type);
+	}
```
Comment (garnermic) [inline meta-facebook/yv35-hd/src/platform/plat_class.c:321]: Add an else statemetn here or change else-if to else if it handles all failure cases.

**Review by garnermic — CHANGES_REQUESTED** [review]
Few minor changes requested.  Thanks!

**Comment by Yi-Shum** [comment]
Hi @garnermic ,
I modified it based on the comments.
Please review again, thank you.

**Comment by Yi-Shum** [comment]
Close this PR since HSC OCP setting will be changed.
Will create another PR to support BOM3 sensor.

**Pull request closed without merging**